ClientApp/viking-vault/src/UserDashboard/EditAutomaticPaymentModal.tsx

```diff
@@ -0,0 +1,181 @@
+                this.setState({
+                    errorLabel: "Error. Please try again later!"
+                });
+                setTimeout(() => {
+                    this.setState({
+                    errorLabel: ""
+                    });
+                }, 2500);
+                return null;
+                }
+                return response.json();
+            })
+            .then(result => {
+                if (this.state.errorLabel === "") {
+                    this.closeModal()
+                }
+            });
+        }
+    }
+
+    render() {
+        return (
+            <div>
+                <div className={this.props.open? "modal open" : "modal close"}>
+                    <div className="container col-md-6">
+                        <div className="modal-dialog attach-card">
+                            <div className="modal-content attach-card">
+                                <div className="modal-header attach-card">
+                                    <h5 className="heading-name attach-card">Edit Automatic Payment Form</h5>
+                                </div>
+                                <div className="form-group attach-card">
+                                    <label className="form-label attach-card">Amount</label>
+                                    <input type="text" value={this.state.automaticPayment.amount} onChange={(e) => this.handleChange(e.target.value, "amount")} required className="form-control attach-card"></input>
+                                    {this.amountMustBePositive().message !== "Ok" ? <pre className={this.amountMustBePositive().class}>{this.amountMustBePositive().message}</pre> : null}
+                                </div>
+                                <div className="form-group attach-card">
+                                    <label className="form-label attach-card">Initial Payment Date</label>
+                                    <input type="date" name="initialPaymentDate" value={this.state.automaticPayment.initialPaymentDate} onChange={(e) => this.handleChange(e.target.value, "initialPaymentDate")} required className="form-control attach-card"></input>
+                                    { this.initialPaymentDateIsAfterCurrentDate().message !== "Ok" ? <pre className={this.initialPaymentDateIsAfterCurrentDate().class}>{this.initialPaymentDateIsAfterCurrentDate().message}</pre> : null }
+                                </div>
+                                {this.state.errorLabel !== "" ? <div className="alert alert-warning"> {this.state.errorLabel} </div> : null }
+                                <div className="modal-footer attach-card">
+                                    <button disabled={!this.mandatoryFieldsCompletedCorrectly()} className={this.mandatoryFieldsCompletedCorrectly()? "btn btn-primary attach-card" : "btn btn-secondary attach-card"} onClick={() => this.sendDataAndShowResponse()}>Edit Automatic Payment</button>
```

Comment:
Separate the props on their own line if the component get too big that you have to scroll horizontally